<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ MItools</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Function reference" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MItools</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1.40</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/articles/Full_interactomes_interspecies.html">Getting started: full interactomes and interspecies interactions</a>
    </li>
    <li>
      <a href="../articles/articles/MItools_vs_igraph.html">MItools vs igraph: calculating degree of a protein</a>
    </li>
    <li>
      <a href="../articles/articles/study_bias_doc.html">Study bias in the protein network</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-all-functions" class="hasAnchor"><a href="#section-all-functions" class="anchor"></a>All functions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="ELMdb2GRanges.html">ELMdb2GRanges()</a></code> </p>
        </td>
        <td><p>Download and read ELM database motif occurrences</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="GRangesINinteractionSubsetFASTA.html">GRangesINinteractionSubsetFASTA()</a></code> </p>
        </td>
        <td><p>Check if proteins in GRanges were used for motif search using QSLIMFinder (SLIMFinder)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MITABregionFeature.html">MITABregionFeature()</a></code> </p>
        </td>
        <td><p>Extract binding region feature from cleaned MITAB2.7</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="PPInetwork2SLIMFinder.html">PPInetwork2SLIMFinder()</a></code> </p>
        </td>
        <td><p>Find linear motifs (QSLIMFinder or SLIMFinder) in the protein interaction network</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="PermutResult2D.html">PermutResult2D()</a></code> </p>
        </td>
        <td><p>2D bin plot of the features of top N domain-protein pairs as identified by <code>permutationPval</code></p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="QSLIMFinderCommand.html">QSLIMFinderCommand()</a></code> <code><a href="QSLIMFinderCommand.html">mQSLIMFinderCommand()</a></code> <code><a href="QSLIMFinderCommand.html">groupQSLIMFinderCommand()</a></code> </p>
        </td>
        <td><p>Generate a QSLIMFinder command</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="SLIMFinderOcc2GRanges.html">SLIMFinderOcc2GRanges()</a></code> </p>
        </td>
        <td><p>Read QSLIMFinder (SLIMFinder) occurence output to Genomic Ranges object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="XYZ.p.adjust.html">XYZ.p.adjust()</a></code> </p>
        </td>
        <td><p><code>XYZ.p.adjust</code> adjusts p.value in the permutation result</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="benchmarkMotifs.html">benchmarkMotifs()</a></code> <code><a href="benchmarkMotifs.html">queryOCCByMCOL()</a></code> <code><a href="benchmarkMotifs.html">mBenchmarkMotifs()</a></code> </p>
        </td>
        <td><p>Benchmark linear motif instance found using QSLIMFinder (SLIMFinder)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="benchmarkMotifsROC.html">benchmarkMotifsROC()</a></code> <code><a href="benchmarkMotifsROC.html">mBenchmarkMotifsROC()</a></code> </p>
        </td>
        <td><p>Plot ROC or precision-recall curves for motifs search benchmark</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="browseMIontology.html">browseMIontology()</a></code> </p>
        </td>
        <td><p>browse Molecular Interaction ontology</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="centerDomains.html">centerDomains()</a></code> </p>
        </td>
        <td><p>Center QSLIMFinder datasets (InteractionSubsetFASTA_list) at domains</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cleanMITAB.html">cleanMITAB()</a></code> </p>
        </td>
        <td><p>Process molecular interaction data in MITAB 2.5 or 2.7 formats</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cleanMITAB25.html">cleanMITAB25()</a></code> </p>
        </td>
        <td><p><code>cleanMITAB</code> extracts and cleans a set of columns of MITAB2.5</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cleanMITAB27.html">cleanMITAB27()</a></code> </p>
        </td>
        <td><p><code>cleanMITAB</code> extracts and cleans a set of columns of MITAB2.7</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="collapseByInterProID.html">collapseByInterProID()</a></code> </p>
        </td>
        <td><p>Collapse InterProScan output to one protein sequence feature per InterProID (or other feature attribute)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="domainEnrichment.html">domainEnrichment()</a></code> </p>
        </td>
        <td><p>domainEnrichment: filter domains by background domain count =&gt; foldEnrichmentDist =&gt; foldEnrichmentPval =&gt; pval correction =&gt; plot pvals =&gt; plot the distribution for a set of random proteins</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="domainProteinPairMatch.html">domainProteinPairMatch()</a></code> </p>
        </td>
        <td><p>Check if the protein in the set2 is associated with domain that is contained in the protein in the name and remove if not</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="downloadFasta.html">downloadFasta()</a></code> <code><a href="downloadFasta.html">loadAllFASTA()</a></code> </p>
        </td>
        <td><p>Download FASTA for a list of canonical (P04637) and isoform (P04637-2) UniprotAC</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="downloadFastaMixed.html">downloadFastaMixed()</a></code> </p>
        </td>
        <td><p>Download FASTA for a list of canonical (P04637) and isoform (P04637-2) UniprotAC</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="downloadFastaPostproc.html">downloadFastaPostproc()</a></code> </p>
        </td>
        <td><p>Download FASTA for a list of post-processed chain ID (e.g. PRO_0000261216)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="edgelist2degree.html">edgelist2degree</a></code> </p>
        </td>
        <td><p>find the degree for each protein given cleaned molecular interaction data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="edgelist2degreeFreq.html">edgelist2degreeFreq()</a></code> </p>
        </td>
        <td><p>Find the degree distribution given edgelist (pair_id) and a list of proteins which degrees matter</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractInteractors.html">extractInteractors()</a></code> <code><a href="extractInteractors.html">uniqueNinteractions()</a></code> <code><a href="extractInteractors.html">uniqueNinteractors()</a></code> </p>
        </td>
        <td><p>Subset MITAB to retrieve interactors, the number of interactions ot interactors</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filterBYmotifType.html">filterBYmotifType()</a></code> </p>
        </td>
        <td><p>Filter ELM database motif occurrence GRanges by motif type</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filterInteractionSubsetFASTA_list.html">filterInteractionSubsetFASTA_list()</a></code> </p>
        </td>
        <td><p>Filter InteractionSubsetFASTA_list by the size of the interaction sets (number of proteins in each)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="findOverlapsBench.html">findOverlapsBench()</a></code> </p>
        </td>
        <td><p>Find occuring GRanges overlapping with the benchmarking set (positive + negative examples)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="foldEnrichment.html">foldEnrichment()</a></code> </p>
        </td>
        <td><p><code>foldEnrichment</code> calculates fold enrichment of a domain (or other feature) among interacting partners of a protein</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="foldEnrichmentDist.html">library()</a></code> <code><a href="foldEnrichmentDist.html">`=`()</a></code> <code><a href="foldEnrichmentDist.html">`=`()</a></code> <code><a href="foldEnrichmentDist.html">`=`()</a></code> <code><a href="foldEnrichmentDist.html">`=`()</a></code> <code><a href="foldEnrichmentDist.html">foldEnrichmentDist()</a></code> </p>
        </td>
        <td><p>find fold_enrichment distribution (in any domain or other feature) for each protein in the network</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="foldEnrichmentPval.html">foldEnrichmentPval()</a></code> <code><a href="foldEnrichmentPval.html">plotFoldEnrichmentDist()</a></code> </p>
        </td>
        <td><p>Find pvalue for each protein-feature pair in the network</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fullInteractome.html">fullInteractome()</a></code> </p>
        </td>
        <td><p>Retrieve interactome (proteins only or not) of a given taxonomic species</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="humanViralDegree.html">humanViralDegree()</a></code> </p>
        </td>
        <td><p>Compare viral and human protein degree distribution, filter by interaction detection method or PMIDs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="interSpeciesInteractome.html">interSpeciesInteractome()</a></code> </p>
        </td>
        <td><p>Retrieve interspecies interactions (proteins only or not) for a given pair of taxonomy IDs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="interactionSubsetFASTA.html">interactionSubsetFASTA()</a></code> </p>
        </td>
        <td><p>FASTA sequences of interactors of a specific protein</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="interactionSubsetMapID.html">interactionSubsetMapID()</a></code> </p>
        </td>
        <td><p>interaction_subset_from_2_sets renames ids in the interaction_subset_from_2_sets object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="interactorFeatureTypes.html">interactorFeatureTypes()</a></code> </p>
        </td>
        <td><p>discover types of interactor features in MITAB 2.7</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="lastIntActRelease.html">lastIntActRelease()</a></code> </p>
        </td>
        <td><p>last IntAct (all IMEx databases) release</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="listInteractionSubsetFASTA.html">listInteractionSubsetFASTA()</a></code> </p>
        </td>
        <td><p>get FASTA sequences of proteins that interact with seed proteins (datasets for QSLIMFinder)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="listSingleInteractFromSet2.html">listSingleInteractFromSet2()</a></code> </p>
        </td>
        <td><p>Split interaction subset from 2 sets so that each elements contains only one of the proteins from set 2, and retrieve FASTA sequences for each</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="loadAllIDProteins.html">loadAllIDProteins()</a></code> </p>
        </td>
        <td><p>load the list of all protein ID in UniProtKB</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="loadBioplex.html">loadBioplex()</a></code> </p>
        </td>
        <td><p>load Bioplex data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="loadHumanViralPPI.html">loadHumanViralPPI()</a></code> </p>
        </td>
        <td><p>load human-human, human-viral and viral-viral PPI data from IntActFTP</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="loadIntActFTP.html">loadIntActFTP()</a></code> </p>
        </td>
        <td><p>load all data from IntAct ftp</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="loadInterProTree.html">loadInterProTree()</a></code> </p>
        </td>
        <td><p>Load the hierarchy of relationships between InterPro's entries</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="loadMIontology.html">loadMIontology()</a></code> </p>
        </td>
        <td><p>load Molecular Interaction ontology</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="loadTaxIDAllLower.html">loadTaxIDAllLower()</a></code> </p>
        </td>
        <td><p>load all lower taxonomy ID (all descendants)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="loadTaxIDProteins.html">loadTaxIDProteins()</a></code> </p>
        </td>
        <td><p>load the list of all proteins for given taxonomy ID (any level)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="merge2GRangesmcols.html">merge2GRangesmcols()</a></code> </p>
        </td>
        <td><p>Merge data.table to GRanges metadata</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="motifVariantClass.html">motifVariantClass()</a></code> </p>
        </td>
        <td><p>Generate motif variant class (the first 2 element in the motif names)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="parReplicate.html">parReplicate()</a></code> </p>
        </td>
        <td><p>parReplicate function from: https://stackoverflow.com/questions/19281010/simplest-way-to-do-parallel-replicate</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="permutationPval.html">`=`()</a></code> <code><a href="permutationPval.html">res</a></code> <code><a href="permutationPval.html">plot()</a></code> <code><a href="permutationPval.html">plot()</a></code> </p>
        </td>
        <td><p>calculate permutation pval for association between X and Z if both are connected to Y</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="permutationPvalHelper.html">permutationPvalHelper</a></code> <code><a href="permutationPvalHelper.html">`=`()</a></code> <code><a href="permutationPvalHelper.html">`=`()</a></code> <code><a href="permutationPvalHelper.html">`=`()</a></code> <code><a href="permutationPvalHelper.html">`=`()</a></code> <code><a href="permutationPvalHelper.html">`=`()</a></code> <code><a href="permutationPvalHelper.html">`=`()</a></code> <code><a href="permutationPvalHelper.html">`=`()</a></code> <code><a href="permutationPvalHelper.html">`=`()</a></code> <code><a href="permutationPvalHelper.html">`=`()</a></code> <code><a href="permutationPvalHelper.html">`=`()</a></code> <code><a href="permutationPvalHelper.html">`=`()</a></code> </p>
        </td>
        <td><p>functions that are called inside permutationPval</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotEnrichment.html">plotEnrichment()</a></code> </p>
        </td>
        <td><p>Plot the enrichment of domains that are likely to be mediating interactions in domain annotated by ELM as mediating linear motif mediated interactions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="printMotifDomainTable.html">printMotifDomainTable()</a></code> </p>
        </td>
        <td><p><code>printMotifDomainTable</code> combines Domain enrichment and Motif search results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="queryPSICQUIC.html">queryPSICQUIC()</a></code> </p>
        </td>
        <td><p>Use PSICQUIC to query for molecular interactions (using MIQL, specifying database and output format)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="queryPSICQUICrlib.html">queryPSICQUICrlib()</a></code> </p>
        </td>
        <td><p>download PSICQUIC query search result (only IMEx databases) to R library - queryPSICQUIC package - data directory</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="randomGRanges.html">randomGRanges()</a></code> </p>
        </td>
        <td><p>Generate random GRanges of equal length in the same sequence</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="randomInteractome.html">randomInteractome()</a></code> </p>
        </td>
        <td><p>Retrieve molecular interactions for the random set of proteins (of a particular taxon)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="readInterProGFF3.html">addInterProEntryTypes()</a></code> </p>
        </td>
        <td><p>Read InterProScan output in the gff format, select sequence features by type</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="readInteractome.html">readInteractome()</a></code> </p>
        </td>
        <td><p>read Interactome into data.table</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="readQSLIMFinderMain.html">readQSLIMFinderMain()</a></code> <code><a href="readQSLIMFinderMain.html">readQSLIMFinderOccurence()</a></code> <code><a href="readQSLIMFinderMain.html">writePatternList()</a></code> </p>
        </td>
        <td><p>Read and write QSLIMFinder results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="recodeANDinteractionSubsetFASTA.html">recodeANDinteractionSubsetFASTA()</a></code> </p>
        </td>
        <td><p>Recode names and retrieve FASTA sequences of interactors of a specific protein</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="recodeFASTA.html">recodeFASTA()</a></code> </p>
        </td>
        <td><p>Recode names of the AAStringSet object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="reduceOverlappingGRanges.html">reduceOverlappingGRanges()</a></code> </p>
        </td>
        <td><p>Reduce Overlapping GRanges keeping metadata (collapsing withing each column separated by pipes)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="reduceWithMcols.html">reduceWithMcols()</a></code> </p>
        </td>
        <td><p>Reduce GRanges keeping metadata (collapsing withing each column separated by pipes)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="removeInteractionNoFASTA.html">removeInteractionNoFASTA()</a></code> </p>
        </td>
        <td><p>Remove interactions if one participant doesn't have FASTA sequence</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="removeInteractionObsoleteID.html">removeInteractionObsoleteID()</a></code> </p>
        </td>
        <td><p>Remove interactions if one participant has obsolete Uniprot accession (ID)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="removeSelfLoops.html">removeSelfLoops()</a></code> </p>
        </td>
        <td><p>Remove self loops (self-interactions)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="reorderMITAB25.html">reorderMITAB25()</a></code> </p>
        </td>
        <td><p><code>reorderMITAB25</code> reorders interacting molecules in a pair (and all the corresponding columns) according to order provided in IDs_A_order and IDs_B_order columns (latter are deleted)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="reorderMITAB27.html">reorderMITAB27()</a></code> </p>
        </td>
        <td><p><code>reorderMITAB27</code> reorders interacting molecules in a pair (and all the corresponding columns) according to order provided in IDs_A_order and IDs_B_order columns (latter are deleted)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runCompariMotif3.html">runCompariMotif3()</a></code> </p>
        </td>
        <td><p>Run CompariMotif3</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runQSLIMFinder.html">runQSLIMFinder()</a></code> <code><a href="runQSLIMFinder.html">modifyMemoryInBsub()</a></code> <code><a href="runQSLIMFinder.html">LSFrunQSLIMFinder()</a></code> <code><a href="runQSLIMFinder.html">jobsCrashed()</a></code> </p>
        </td>
        <td><p>Run a list of QSLIMFinder jobs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="singleInteractFromSet2.html">singleInteractFromSet2()</a></code> </p>
        </td>
        <td><p>Filter interaction subset from 2 sets: keep only i-th element from the set2</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="subset2setsBy1ID.html">subset2setsBy1ID()</a></code> </p>
        </td>
        <td><p>Subset 2 sets of protein interaction data using a single ID and combine the results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="subsetMITABbyID.html">subsetMITABbyID()</a></code> </p>
        </td>
        <td><p>Retrieve interactions for a list of interactors (given data table, not using webservice)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="subsetMITABbyMethod.html">subsetMITABbyMethod()</a></code> </p>
        </td>
        <td><p>Filter molecular interactions by interaction detection methods and / or participant identification method</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="subsetMITABbyPMIDs.html">subsetMITABbyPMIDs()</a></code> </p>
        </td>
        <td><p>Filter molecular interactions by Pubmed ID</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="writeInteractionSubsetFASTA_list.html">writeInteractionSubsetFASTA_list()</a></code> </p>
        </td>
        <td><p>Write InteractionSubsetFASTA_list into directory system (creating directories as necessary) and return file locations for QSLIMFinder</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#section-all-functions">All functions</a></li>
    </ul>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Vitalii Kleshchevnikov.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

