<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Find linear motifs (QSLIMFinder or SLIMFinder) in the protein interaction network — PPInetwork2SLIMFinder • MItools</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Find linear motifs (QSLIMFinder or SLIMFinder) in the protein interaction network — PPInetwork2SLIMFinder" />

<meta property="og:description" content="Find linear motifs (QSLIMFinder or SLIMFinder) in the protein interaction network" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MItools</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1.40</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/articles/Full_interactomes_interspecies.html">Getting started: full interactomes and interspecies interactions</a>
    </li>
    <li>
      <a href="../articles/articles/MItools_vs_igraph.html">MItools vs igraph: calculating degree of a protein</a>
    </li>
    <li>
      <a href="../articles/articles/study_bias_doc.html">Study bias in the protein network</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Find linear motifs (QSLIMFinder or SLIMFinder) in the protein interaction network</h1>
    
    <div class="hidden name"><code>PPInetwork2SLIMFinder.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Find linear motifs (QSLIMFinder or SLIMFinder) in the protein interaction network</p>
    
    </div>

    <pre class="usage"><span class='fu'>PPInetwork2SLIMFinder</span>(<span class='kw'>dataset_name</span> <span class='kw'>=</span> <span class='st'>"SLIMFinder"</span>,
  <span class='kw'>interaction_main_set</span> <span class='kw'>=</span> <span class='no'>all_human_interaction</span>,
  <span class='kw'>interaction_query_set</span> <span class='kw'>=</span> <span class='no'>all_viral_interaction</span>,
  <span class='kw'>analysis_type</span> <span class='kw'>=</span> <span class='st'>"qslimfinder"</span>,
  <span class='kw'>options</span> <span class='kw'>=</span> <span class='st'>"dismask=T consmask=F cloudfix=T probcut=0.3 minwild=0 maxwild=2 slimlen=5 alphahelix=F maxseq=1500 savespace=1 iuchdir=T"</span>,
  <span class='kw'>domain_res_file</span> <span class='kw'>=</span> <span class='st'>"./processed_data_files/what_we_find_VS_ELM_clust20171019.RData"</span>,
  <span class='kw'>domain_results_obj</span> <span class='kw'>=</span> <span class='st'>"res_count"</span>, <span class='kw'>center_domains</span> <span class='kw'>=</span> <span class='no'>F</span>,
  <span class='kw'>fasta_path</span> <span class='kw'>=</span> <span class='st'>"./data_files/all_human_viral_proteins.fasta"</span>,
  <span class='kw'>main_set_only</span> <span class='kw'>=</span> <span class='no'>F</span>, <span class='kw'>domain_pvalue_cutoff</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>SLIMFinder_dir</span> <span class='kw'>=</span> <span class='fu'>paste0</span>(<span class='st'>"./"</span>, <span class='no'>dataset_name</span>, <span class='st'>"/"</span>),
  <span class='kw'>LSF_project_path</span> <span class='kw'>=</span> <span class='st'>"/hps/nobackup/research/petsalaki/users/vitalii/vitalii/viral_project/"</span>,
  <span class='kw'>software_path</span> <span class='kw'>=</span> <span class='st'>"../software/cluster/"</span>, <span class='kw'>length_set1_min</span> <span class='kw'>=</span> <span class='fl'>2</span>,
  <span class='kw'>length_set2_min</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>write_log</span> <span class='kw'>=</span> <span class='no'>T</span>, <span class='kw'>N_seq</span> <span class='kw'>=</span> <span class='fl'>200</span>,
  <span class='kw'>seed_list</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>query_list</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>memory_start</span> <span class='kw'>=</span> <span class='fl'>350</span>,
  <span class='kw'>memory_step</span> <span class='kw'>=</span> <span class='fl'>100</span>, <span class='kw'>compare_motifs</span> <span class='kw'>=</span> <span class='no'>T</span>, <span class='kw'>Njobs_limit</span> <span class='kw'>=</span> <span class='fl'>490</span>,
  <span class='kw'>CompariMotif3_dburl</span> <span class='kw'>=</span> <span class='st'>"http://elm.eu.org/elms/elms_index.tsv"</span>,
  <span class='kw'>CompariMotif3_dbpath</span> <span class='kw'>=</span> <span class='st'>"./data_files/"</span>,
  <span class='kw'>non_query_domain_res_file</span> <span class='kw'>=</span> <span class='st'>"../viral_project/processed_data_files/predict_domain_human_clust20180819.RData"</span>,
  <span class='kw'>non_query_domain_results_obj</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>non_query_domains_N</span> <span class='kw'>=</span> <span class='fl'>0</span>,
  <span class='kw'>non_query_set_only</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='no'>main_set_only</span>), <span class='kw'>query_domains_only</span> <span class='kw'>=</span> <span class='no'>T</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>dataset_name</th>
      <td><p>refer to <code><a href='benchmarkMotifs.html'>mBenchmarkMotifs</a></code></p></td>
    </tr>
    <tr>
      <th>interaction_main_set</th>
      <td><p>clean_MItab class, use this set of protein interactions to construct QSLIMFinder datasets</p></td>
    </tr>
    <tr>
      <th>interaction_query_set</th>
      <td><p>clean_MItab class, use this set of protein interactions as a query (+ add to the QSLIMFinder datasets). Both interaction sets have shared seed proteins. SLIMFinder <code>analysis_type</code> also requires this option because it add proteins from these interactions to the SLIMFinder datasets</p></td>
    </tr>
    <tr>
      <th>analysis_type</th>
      <td><p>"qslimfinder" or "slimfinder"</p></td>
    </tr>
    <tr>
      <th>options</th>
      <td><p>any options from QSLIMFinder or SLIMFinder. Detail http://rest.slimsuite.unsw.edu.au/docs&amp;page=module:qslimfinder or http://rest.slimsuite.unsw.edu.au/docs&amp;page=module:slimfinder =&gt; Commandline</p></td>
    </tr>
    <tr>
      <th>domain_res_file</th>
      <td><p>relative path to domain enrichment results RData</p></td>
    </tr>
    <tr>
      <th>domain_results_obj</th>
      <td><p>which object contains domain enrichment results in <code>domain_res_file</code>, XYZinteration_XZEmpiricalPval?</p></td>
    </tr>
    <tr>
      <th>center_domains</th>
      <td><p>logical, center QSLIMFinder datasets at domains?</p></td>
    </tr>
    <tr>
      <th>fasta_path</th>
      <td><p>relative path (from the project folder) to the FASTA file containing sequences for all proteins in <code>interaction_main_set</code> and <code>interaction_query_set</code></p></td>
    </tr>
    <tr>
      <th>main_set_only</th>
      <td><p>logical, If TRUE sequence sets for motif search contain only proteins from <code>interaction_main_set</code>. If FALSE, non-query proteins from <code>interaction_query_set</code> are also included. Argument for <code><a href='listInteractionSubsetFASTA.html'>listInteractionSubsetFASTA</a></code></p></td>
    </tr>
    <tr>
      <th>domain_pvalue_cutoff</th>
      <td><p>construct SLIMFinder datasets using interactions of proteins that contain domain associated to protein in the query set with p-value <code>domain_pvalue_cutoff</code> or lower</p></td>
    </tr>
    <tr>
      <th>SLIMFinder_dir</th>
      <td><p>directory to store SLIMFinder datasets and results within the project directory</p></td>
    </tr>
    <tr>
      <th>LSF_project_path</th>
      <td><p>full path to the project directory</p></td>
    </tr>
    <tr>
      <th>software_path</th>
      <td><p>relative path (from the project folder) to the directory containing slimsuite, blast, iupred # "../software/cluster/" or "../software/"</p></td>
    </tr>
    <tr>
      <th>length_set1_min</th>
      <td><p>mininal number of proteins in a QSLIMFinder dataset from <code>interaction_main_set</code>. Argument for <code><a href='filterInteractionSubsetFASTA_list.html'>filterInteractionSubsetFASTA_list</a></code></p></td>
    </tr>
    <tr>
      <th>length_set2_min</th>
      <td><p>mininal number of proteins in a QSLIMFinder dataset from <code>interaction_query_set</code>. Argument for <code><a href='filterInteractionSubsetFASTA_list.html'>filterInteractionSubsetFASTA_list</a></code></p></td>
    </tr>
    <tr>
      <th>write_log</th>
      <td><p>FALSE will not allow runQSLIMFinder to detect crashed jobs</p></td>
    </tr>
    <tr>
      <th>N_seq</th>
      <td><p>number of sequences per batch</p></td>
    </tr>
    <tr>
      <th>seed_list</th>
      <td><p>character vector of UniprotKB accesions that should serve as a seed for QSLIMFinder datasets. These proteins are supposed to recognise SLIMs. Overrides selection of seed protein by <code>domain_pvalue_cutoff</code></p></td>
    </tr>
    <tr>
      <th>query_list</th>
      <td><p>character vector of UniprotKB accesions that should serve as a query for QSLIMFinder</p></td>
    </tr>
    <tr>
      <th>memory_start</th>
      <td><p>integer, how much memory each job should be given initially</p></td>
    </tr>
    <tr>
      <th>memory_step</th>
      <td><p>interger, increment by which to increase how much memory each job should be given if <code>memory_start</code> is not enough and the job has failed</p></td>
    </tr>
    <tr>
      <th>compare_motifs</th>
      <td><p>logical, compare motifs using CompariMotif3? The procedure is relatively fast but memory consuming.</p></td>
    </tr>
    <tr>
      <th>Njobs_limit</th>
      <td><p>integer, the number of LSF jobs allowed to run simultaneously</p></td>
    </tr>
    <tr>
      <th>CompariMotif3_dburl</th>
      <td><p>dburl url where to download database for CompariMotif V3. Argument for <code><a href='runCompariMotif3.html'>runCompariMotif3</a></code></p></td>
    </tr>
    <tr>
      <th>CompariMotif3_dbpath</th>
      <td><p>path to directory where to save and keep ELM database (http://elm.eu.org/) or other database of linear motifs in a format required by comparimotif_V3: http://rest.slimsuite.unsw.edu.au/docs&amp;page=module:comparimotif_V3</p></td>
    </tr>
    <tr>
      <th>non_query_domain_res_file</th>
      <td><p>path to RData file containing the result of domain enrichment analysis for non-query proteins</p></td>
    </tr>
    <tr>
      <th>non_query_domain_results_obj</th>
      <td><p>character, name of the object containing domain enrichment results for non-query proteins (class == XYZinteration_XZEmpiricalPval), when provided will be used for filtering datasets.</p></td>
    </tr>
    <tr>
      <th>non_query_domains_N</th>
      <td><p>the number of non-query proteins with predicted domains for each dataset. Used only when non_query_domain_results_obj is not NULL</p></td>
    </tr>
    <tr>
      <th>non_query_set_only</th>
      <td><p>If TRUE sequence sets for motif search contain only proteins (interacting partners of a seed) from non_query_domain_results_obj, if FALSE - both from non_query_domain_results_obj and domain_res_obj. Used only when non_query_domain_results_obj is not NULL and by default equals to main_set_only</p></td>
    </tr>
    <tr>
      <th>query_domains_only</th>
      <td><p>If TRUE proteins whose sequences will be used for motif search must be predicted to bind the same domains in a seed protein as domains predicted for query protein. Used only when non_query_domain_results_obj is not NULL</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>path to RData containing all objects used by this pipeline</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>QSLIMFinder command line options (http://rest.slimsuite.unsw.edu.au/docs&amp;page=module:qslimfinder)</p>
<p>### Basic Input/Output Options ###</p>
<p>seqin=FILE : Sequence file to search [None]</p>
<p>batch=LIST : List of files to search, wildcards allowed. (Over-ruled by seqin=FILE.) [*.dat,*.fas]</p>
<p>query=LIST : Return only SLiMs that occur in 1+ Query sequences (Name/AccNum/Seq Number) [1]</p>
<p>addquery=FILE : Adds query sequence(s) to batch jobs from FILE [None]</p>
<p>maxseq=X : Maximum number of sequences to process [500]</p>
<p>maxupc=X : Maximum UPC size of dataset to process [0]</p>
<p>sizesort=X : Sorts batch files by size prior to running (+1 small-&gt;big; -1 big-&gt;small; 0 none) [0]</p>
<p>walltime=X : Time in hours before program will abort search and exit [1.0]</p>
<p>resfile=FILE : Main QSLiMFinder results table [qslimfinder.csv]</p>
<p>resdir=PATH : Redirect individual output files to specified directory (and look for intermediates) [QSLiMFinder/]</p>
<p>buildpath=PATH : Alternative path to look for existing intermediate files [SLiMFinder/]</p>
<p>force=T/F : Force re-running of BLAST, UPC generation and SLiMBuild [False]</p>
<p>pickup=T/F : Pick-up from aborted batch run by identifying datasets in resfile using RunID [False]</p>
<p>dna=T/F : Whether the sequences files are DNA rather than protein [False]</p>
<p>alphabet=LIST : List of characters to include in search (e.g. AAs or NTs) [default AA or NT codes]</p>
<p>megaslim=FILE : Make/use precomputed results for a proteome (FILE) in fasta format [None]</p>
<p>megablam=T/F : Whether to create and use all-by-all GABLAM results for (gablamdis) UPC generation [False]</p>
<p>ptmlist=LIST : List of PTM letters to add to alphabet for analysis and restrict PTM data []</p>
<p>ptmdata=DSVFILE : File containing PTM data, including AccNum, ModType, ModPos, ModAA, ModCode</p>
<p>#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~#</p>
<p>SLiMBuild Options I</p>
<p>efilter=T/F : Whether to use evolutionary filter [True]</p>
<p>blastf=T/F : Use BLAST Complexity filter when determining relationships [True]</p>
<p>blaste=X : BLAST e-value threshold for determining relationships [1e=4]</p>
<p>altdis=FILE : Alternative all by all distance matrix for relationships [None]</p>
<p>gablamdis=FILE : Alternative GABLAM results file [None] (!!!Experimental feature!!!)</p>
<p>homcut=X : Max number of homologues to allow (to reduce large multi-domain families) [0]</p>
<p>SLiMBuild Options II</p>
<p>masking=T/F : Master control switch to turn off all masking if False [True]</p>
<p>dismask=T/F : Whether to mask ordered regions (see rje_disorder for options) [False]</p>
<p>consmask=T/F : Whether to use relative conservation masking [False]</p>
<p>ftmask=LIST : UniProt features to mask out (True=EM,DOMAIN,TRANSMEM) []</p>
<p>imask=LIST : UniProt features to inversely ("inclusively") mask. (Seqs MUST have 1+ features) []</p>
<p>compmask=X,Y : Mask low complexity regions (same AA in X+ of Y consecutive aas) [5,8]</p>
<p>casemask=X : Mask Upper or Lower case [None]</p>
<p>motifmask=X : List (or file) of motifs to mask from input sequences []</p>
<p>metmask=T/F : Masks the N-terminal M (can be useful if termini=T) [True]</p>
<p>posmask=LIST : Masks list of position-specific aas, where list = pos1:aas,pos2:aas [2:A]</p>
<p>aamask=LIST : Masks list of AAs from all sequences (reduces alphabet) []</p>
<p>qregion=X,Y : Mask all but the region of the query from (and including) residue X to residue Y [0,-1]</p>
<p>SLiMBuild Options III</p>
<p>termini=T/F : Whether to add termini characters (^ &amp; $) to search sequences [True]</p>
<p>minwild=X : Minimum number of consecutive wildcard positions to allow [0]</p>
<p>maxwild=X : Maximum number of consecutive wildcard positions to allow [2]</p>
<p>slimlen=X : Maximum length of SLiMs to return (no. non-wildcard positions) [5]</p>
<p>minocc=X : Minimum number of unrelated occurrences for returned SLiMs. (Proportion of UP if &lt; 1) [0.05]</p>
<p>absmin=X : Used if minocc&lt;1 to define absolute min. UP occ [3]</p>
<p>alphahelix=T/F : Special i, i+3/4, i+7 motif discovery [False]</p>
<p>SLiMBuild Options IV</p>
<p>ambiguity=T/F : (preamb=T/F) Whether to search for ambiguous motifs during motif discovery [True]</p>
<p>ambocc=X : Min. UP occurrence for subvariants of ambiguous motifs (minocc if 0 or &gt; minocc) [0.05]</p>
<p>absminamb=X : Used if ambocc&lt;1 to define absolute min. UP occ [2]</p>
<p>equiv=LIST : List (or file) of TEIRESIAS-style ambiguities to use [AGS,ILMVF,FYW,FYH,KRH,DE,ST]</p>
<p>wildvar=T/F : Whether to allow variable length wildcards [True]</p>
<p>combamb=T/F : Whether to search for combined amino acid degeneracy and variable wildcards [False]</p>
<p>SLiMBuild Options V</p>
<p>musthave=LIST : Returned motifs must contain one or more of the AAs in LIST (reduces search space) []</p>
<p>focus=FILE : FILE containing focal groups for SLiM return (see Manual for details) [None]</p>
<p>focusocc=X : Motif must appear in X+ focus groups (0 = all) [0]</p>
<p>* See also rje_slimcalc options for occurrence-based calculations and filtering *</p>
<p>#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~#</p>
<p>### SLiMChance Options ###</p>
<p>cloudfix=T/F : Restrict output to clouds with 1+ fixed motif (recommended) [False]</p>
<p>slimchance=T/F : Execute main QSLiMFinder probability method and outputs [True]</p>
<p>sigprime=T/F : Calculate more precise (but more computationally intensive) statistical model [False]</p>
<p>sigv=T/F : Use the more precise (but more computationally intensive) fix to mean UPC probability [False]</p>
<p>qexact=T/F : Calculate exact Query motif space (True) or over-estimate from dimers (False) (quicker) [True]</p>
<p>probcut=X : Probability cut-off for returned motifs [0.1]</p>
<p>maskfreq=T/F : Whether to use masked AA Frequencies (True), or (False) mask after frequency calculations [False]</p>
<p>aafreq=FILE : Use FILE to replace individual sequence AAFreqs (FILE can be sequences or aafreq) [None]</p>
<p>aadimerfreq=FILE: Use empirical dimer frequencies from FILE (fasta or *.aadimer.tdt) (!!!Experimental!!!) [None]</p>
<p>negatives=FILE : Multiply raw probabilities by under-representation in FILE (!!!Experimental!!!) [None]</p>
<p>smearfreq=T/F : Whether to "smear" AA frequencies across UPC rather than keep separate AAFreqs [False]</p>
<p>seqocc=T/F : Whether to upweight for multiple occurrences in same sequence (heuristic) [False]</p>
<p>probscore=X : Score to be used for probability cut-off and ranking (Prob/Sig) [Sig]</p>
<p>#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~#</p>
<p>Advanced Output Options I</p>
<p>clouds=X : Identifies motif "clouds" which overlap at 2+ positions in X+ sequences (0=minocc / -1=off) [2]</p>
<p>runid=X : Run ID for resfile (allows multiple runs on same data) [DATE:TIME]</p>
<p>logmask=T/F : Whether to log the masking of individual sequences [True]</p>
<p>slimcheck=FILE : Motif file/list to add to resfile output []</p>
<p>Advanced Output Options II</p>
<p>teiresias=T/F : Replace TEIRESIAS, making *.out and *.mask.fasta files [False]</p>
<p>slimdisc=T/F : Emulate SLiMDisc output format (*.rank &amp; *.dat.rank + TEIRESIAS *.out &amp; *.fasta) [False]</p>
<p>extras=X : Whether to generate additional output files (alignments etc.) [1]</p>
<p>--1 = No output beyond main results file</p>
<p>- 0 = Generate occurrence file and cloud file</p>
<p>- 1 = Generate occurrence file, alignments and cloud file</p>
<p>- 2 = Generate all additional QSLiMFinder outputs</p>
<p>- 3 = Generate SLiMDisc emulation too (equiv extras=2 slimdisc=T)</p>
<p>targz=T/F : Whether to tar and zip dataset result files (UNIX only) [False]</p>
<p>savespace=0 : Delete "unneccessary" files following run (best used with targz): [0]</p>
<p>- 0 = Delete no files</p>
<p>- 1 = Delete all bar *.upc and *.pickle</p>
<p>- 2 = Delete all bar *.upc (pickle added to tar)</p>
<p>- 3 = Delete all dataset-specific files including *.upc and *.pickle (not *.tar.gz)</p>
<p>Advanced Output Options III</p>
<p>topranks=X : Will only output top X motifs meeting probcut [1000]</p>
<p>minic=X : Minimum information content for returned motifs [2.1]</p>
<p>allsig=T/F : Whether to also output all SLiMChance combinations (Sig/SigV/SigPrime/SigPrimeV) [False]</p>
<p>Memory requirements for jobs. How much memory is enough for most jobs?
files = list.files("./qslimfinder.Full_IntAct.FALSE/log_dir/log/")
times = sapply(files, function(file) system(paste0("cat ./qslimfinder.Full_IntAct.FALSE/log_dir/log/",file," | grep Requested"), intern = T))
&gt; table(times)
times
100.00 MB  200.00 MB  300.00 MB  400.00 MB  500.00 MB
14         61         53         25          3
&gt; table(times) / sum(table(times))
times
100.00 MB  200.00 MB  300.00 MB  400.00 MB  500.00 MB
0.08974359 0.39102564 0.33974359 0.16025641 0.01923077
&gt; cumsum(table(times) / sum(table(times)))
100.00 MB  200.00 MB  300.00 MB  400.00 MB  500.00 MB
0.08974359 0.48076923 0.82051282 0.98076923 1.00000000
&gt; 1 - cumsum(table(times) / sum(table(times)))
100.00 MB  200.00 MB  300.00 MB  400.00 MB  500.00 MB
0.91025641 0.51923077 0.17948718 0.01923077 0.00000000</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='listInteractionSubsetFASTA.html'>listInteractionSubsetFASTA</a></code>, <code><a href='runQSLIMFinder.html'>runQSLIMFinder</a></code>, <code><a href='QSLIMFinderCommand.html'>groupQSLIMFinderCommand</a></code>, <code><a href='QSLIMFinderCommand.html'>mQSLIMFinderCommand</a></code>, <code><a href='runCompariMotif3.html'>runCompariMotif3</a></code>, <code><a href='readQSLIMFinderMain.html'>readQSLIMFinderMain</a></code>, <code><a href='readQSLIMFinderMain.html'>readQSLIMFinderOccurence</a></code>, <code><a href='writeInteractionSubsetFASTA_list.html'>writeInteractionSubsetFASTA_list</a></code>, <code><a href='domainProteinPairMatch.html'>domainProteinPairMatch</a></code>, <code><a href='filterInteractionSubsetFASTA_list.html'>filterInteractionSubsetFASTA_list</a></code>, <code><a href='removeInteractionNoFASTA.html'>removeInteractionNoFASTA</a></code>, <code><a href='centerDomains.html'>centerDomains</a></code></p></div>
    

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>

      <li><a href="#see-also">See also</a></li>
          </ul>

    <h2>Author</h2>
    
Vitalii Kleshchevnikov

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Vitalii Kleshchevnikov.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

